### PR TITLE
Case 21147, Case 20944: Fix setTextures (pal, etc.)

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/21147/People-app-Avatar-highlights-are-broken-in-people-app-in-master
https://highfidelity.manuscript.com/f/cases/20944/Fix-Model-textures-property

Test plan:
- Pal should work, no white squares
- Script in 20944 should work.

### CODE REVIEW

File: libraries/model-networking/src/model-networking/TextureCache.cpp [hunk 1]
```diff
@@ -365,7 +367,12 @@ void NetworkTexture::setExtra(void* extra) {
     _type = textureExtra ? textureExtra->type : image::TextureUsage::DEFAULT_TEXTURE;
     _maxNumPixels = textureExtra ? textureExtra->maxNumPixels : ABSOLUTE_MAX_TEXTURE_NUM_PIXELS;
 
-    _textureSource = std::make_shared<gpu::TextureSource>(_url, (int)_type);
+    if (_textureSource) {
```
Comment (SamGondelman) [inline libraries/model-networking/src/model-networking/TextureCache.cpp]: This was the first part of the bug, we were overwriting the _textureSource even if it was already valid.

File: libraries/networking/src/ResourceCache.cpp [hunk 2]
```diff
@@ -489,9 +488,13 @@ void ResourceCache::resetResourceCounters() {
     emit dirty();
 }
 
-void ResourceCache::removeResource(const QUrl& url, qint64 size) {
+void ResourceCache::removeResource(const QUrl& url, size_t extraHash, qint64 size) {
     QWriteLocker locker(&_resourcesLock);
-    _resources.remove(url);
```
Comment (SamGondelman) [inline libraries/networking/src/ResourceCache.cpp]: Second part of bug, this was just wrong.

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17260/HighFidelity-Beta-PR14913-9e002c5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17260/HighFidelity-Beta-Interface-PR14913-9e002c5.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17261/HighFidelity-Beta-PR14913-9e002c5.apk

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17263/HighFidelity-Beta-PR14913-9e002c5.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17263/HighFidelity-Beta-PR14913-9e002c5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17263/HighFidelity-Beta-Interface-PR14913-9e002c5.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17263/HighFidelity-Beta-PR14913-9e002c5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17263/HighFidelity-Beta-Interface-PR14913-9e002c5.exe

**Comment by AndrewMeadows** [comment]
I tested case 21147 and case 20944 and found them fixed in this PR and bugged in hifi/master.  I will now merge this and resolve the cases.

**Pull request merged**